subprojects/gst-plugins-base/gst-libs/gst/gl/gstglupload.c

```diff
@@ -2385,6 +2412,10 @@ gst_gl_upload_transform_caps (GstGLUpload * upload, GstGLContext * context,
         upload_methods[i]->transform_caps (upload->priv->upload_impl[i],
         context, direction, caps);
 
+    if (GST_IS_CAPS (tmp2) && gst_egl_caps_has_feature (tmp2, GST_CAPS_FEATURE_MEMORY_DMABUF)) {
```

Comment:
The DMA handle here is too generic. Not all the "upload" need to check this. You can write a function and let the ones who needs this(for example, _dma_buf_upload_transform_caps) to call inside it.